nominal/sdk.py

```diff
@@ -458,6 +560,59 @@ def search_runs(
         for run in self._search_runs_paginated(request):
             yield Run._from_conjure(self, run)
 
+    def create_log_set(
+        self,
+        name: str,
+        timestamp_type: str,
+        description: str | None = None,
+        origin_metadata: Mapping[str, str] = MappingProxyType({}),
+    ) -> LogSetMetadata:
+        """
+        Creates a log set, to which logs can be attached using `attach-and-finalize`. The logs within a logset are
+        not searchable until the logset is finalized.
+
+        Timestamp type must be a string equal to either 'ABSOLUTE' or 'RELATIVE'.
+        """
+        timestamp_type_enum = None
+        if timestamp_type == "ABSOLUTE":
+            timestamp_type_enum = DataSourceTimestampType.ABSOLUTE
+        elif timestamp_type_enum == "RELATIVE":
+            timestamp_type_enum = DataSourceTimestampType.RELATIVE
+        else:
+            raise TypeError(f"timestamp type {timestamp_type} must be one of [RELATIVE, ABSOLUTE]")
+
+        request = datasource_logset_api.CreateLogSetRequest(
+            name=name,
+            description=description,
+            origin_metadata=None if origin_metadata is None else dict(origin_metadata),
+            timestamp_type=timestamp_type_enum.to_conjure(),
+        )
+        response = self._logset_client.create(self._auth_header, request)
+        return LogSetMetadata._from_conjure(self, response)
+
+    def get_log_set_metadata(self, log_set_rid) -> LogSetMetadata:
+        """Returns metadata about a log set given its RID."""
+        response = self._logset_client.get_log_set_metadata(self._auth_header, log_set_rid)
+        return LogSetMetadata._from_conjure(self, response)
+
+    def search_logs(
```

Comment:
what's the convention on what to put under the Client vs under the LogSetMetadata object itself?